### PR TITLE
Unroll loops in light_compute

### PR DESCRIPTION
**Pull request by Vurich**
Godbolt was showing that this loop (or rather, a representative facsimile of this loop) was not being unrolled, so I used an unrolling macro to force the unrolling to happen. This allows better const-folding and avoids jumps (also, since we make a bunch of function calls that stomp on the registers it avoids having to load from the stack). This is a micro-optimization but benchmarks show almost a 10% improvement.

### CODE REVIEW

File: ethash/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![cfg_attr(feature = "benches", feature(test))]
```
Comment (rphmeier) [inline ethash/src/lib.rs]: place this under the license header

**Comment by rphmeier** [comment]
LGTM other than style grumble. It would be nice to have the `crunchy` crate under the `paritytech:core-devs` group as authors.

File: ethash/src/lib.rs (hunk 1 above)
Comment (rphmeier) [inline ethash/src/lib.rs]: also under the docs :)

**Comment by Vurich** [comment]
The grumbles on this PR have been fixed, can this get a rereview?

File: ethash/src/lib.rs (hunk 1 above)
Comment (rphmeier) [inline ethash/src/lib.rs]: @Vurich this grumble is still an issue

**Comment by rphmeier** [comment]
LGTM modulo the style nit

File: ethash/src/lib.rs (hunk 1 above)
Comment (Vurich) [inline ethash/src/lib.rs]: Oh, sorry. I missed this.

**Comment by Vurich** [comment]
De-nitted

**Pull request merged**